File: be/src/main/java/org/example/gogoma/domain/usermarathon/dto/UserMarathonSearchDto.java
package org.example.gogoma.domain.usermarathon.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class UserMarathonSearchDto {
    private int userMarathonId;
    private String marathonTitle;
    private int marathonType;
    private String dDay;
    private LocalDateTime raceStartDateTime;
    private LocalDateTime paymentDateTime;

    public static UserMarathonSearchDto of(int userMarathonId,
                                           String marathonTitle, int marathonType, String dDay,
            LocalDateTime raceStartDateTime, LocalDateTime paymentDateTime) {

        return UserMarathonSearchDto.builder()
                .userMarathonId(userMarathonId)
                .marathonTitle(marathonTitle)
                .marathonType(marathonType)
                .dDay(dDay)
                .raceStartDateTime(raceStartDateTime)
                .paymentDateTime(paymentDateTime)
                .build();
    }
}

File: app/src/main/java/com/example/gogoma/data/dto/UserMarathonSearchDto.kt
package com.example.gogoma.data.dto

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class UserMarathonSearchDto (
        val userMarathonId: Int,

        val marathonTitle: String,

        val marathonType: String,

        val dDay: String,

        val raceStartDateTime: String,

        val paymentDateTime: String
) : Parcelable